src/Bounds.cpp

```diff
@@ -1368,6 +1368,39 @@ class BoxesTouched : public IRGraphVisitor {
 
     using IRGraphVisitor::visit;
 
+    bool box_from_extended_crop(Expr e, Box &b) {
+        const Call *call_expr = e.as<Call>();
+        if (call_expr != nullptr) {
+            if (call_expr->name == Call::buffer_crop && call_expr->args.size() == 5) {
+                const Variable *in_buf = call_expr->args[2].as<Variable>();
+                const Call *mins_struct = call_expr->args[3].as<Call>();
+                const Call *extents_struct = call_expr->args[4].as<Call>();
+                if (in_buf != nullptr && mins_struct != nullptr && extents_struct != nullptr &&
+                    (in_buf->name == (func + ".buffer")) &&
+                    mins_struct->name == Call::make_struct && extents_struct->name == Call::make_struct) {
+                    b.resize(mins_struct->args.size());
+                    b.used = const_true();
+                    for (size_t i = 0; i < mins_struct->args.size(); i++) {
+                        Interval min_interval = bounds_of_expr_in_scope(mins_struct->args[i], scope, func_bounds);
+                        Interval max_interval = bounds_of_expr_in_scope(mins_struct->args[i] + extents_struct->args[i] - 1, scope, func_bounds);
+                        b[i] = Interval(min_interval.min, max_interval.max);
+                    }
+                    return true;
+                }
+            } else if (call_expr->name == Call::buffer_set_bounds && call_expr->args.size() == 4) {
```

Comment:
I'm concerned the buffer crop gets silently skipped if the args are not as expected. E.g. what happens if one of the mins or extent structs gets pulled out into an outer LetStmt in some future change? Might be safer to assert-fail.